src/main.rs

```diff
@@ -819,20 +872,41 @@ fn backtrace(
         }
 
         let lr = registers.get(LR)?;
-        log::debug!("lr=0x{:08x} pc=0x{:08x}", lr, pc);
+
         if lr == LR_END {
             break;
         }
 
+        let exception_entry = lr > 0xffff_ffe0;
+        let stack_corrupted = !cfa_changed && lr & !THUMB_BIT == pc & !THUMB_BIT;
```

Comment:
Is it possible to clarify what this line expresses with a comment?

Reply:
Do you mean something like `// check for conditions under which we definitely need to print a backtrace` ?

Reply:
You can move the comment from below to the `stack_corrupted` variable:

```rust
// If the frame didn't move, and the program counter didn't change, bail out (otherwise we
// might print the same frame over and over).
// Since we strip the thumb bit from `pc`, ignore it in this comparison.
```

Also this might be insufficient and cause https://github.com/knurling-rs/probe-run/issues/127, so if you'd like to look into that later, this condition would be where I'd start.